### PR TITLE
Allow non-string Resource label values.

### LINKED ISSUE #158: Resource should support the same attribute types as span attributes
I think this is just a documentation change in [Resource](https://github.com/open-telemetry/opentelemetry-ruby/blob/master/sdk/lib/opentelemetry/sdk/resources/resource.rb) to match [Span#set_attribute](https://github.com/open-telemetry/opentelemetry-ruby/blob/master/api/lib/opentelemetry/trace/span.rb).

Addresses https://github.com/open-telemetry/opentelemetry-specification/pull/356

### PR DESCRIPTION
**Pull request by fhwang**
Fixes #158.

### CODE REVIEW

**Comment by fhwang** [comment]
I think I'm set up with the CLA -- commenting here in hopes this will fire the Github check ...

**Comment by fhwang** [comment]
test comment

**Comment by fhwang** [comment]
@mwear any thoughts on how I can fix the CLA issue? I thought I went through the required steps but I probably missed something.

**Comment by elskwid** [comment]
@fhwang I believe you need to post a comment that says `I signed it`

**Comment by fhwang** [comment]
I signed it

File: sdk/lib/opentelemetry/sdk/resources/resource.rb [hunk 1]
```diff
@@ -21,9 +21,10 @@ class << self
           # @return [Resource]
           def create(labels = {})
             frozen_labels = labels.each_with_object({}) do |(k, v), memo|
-              raise ArgumentError, 'label keys and values must be strings' unless k.is_a?(String) && v.is_a?(String)
+              raise ArgumentError, 'label keys must be strings' unless k.is_a?(String)
+              raise ArgumentError, 'label values must be strings, integers, floats, or booleans' unless Internal.valid_value?(v)
```
Comment (fbogsany) [inline sdk/lib/opentelemetry/sdk/resources/resource.rb]: This requires a documentation change as well (the `@param` in the comment above).

**Comment by fhwang** [comment]
I signed it

File: sdk/lib/opentelemetry/sdk/resources/resource.rb [hunk 2]
```diff
@@ -15,15 +15,16 @@ class << self
 
           # Returns a newly created {Resource} with the specified labels
           #
-          # @param [Hash<String, String>] labels Hash of key-value pairs to be used
+          # @param [Hash{String=>String, Numeric, Boolean>} labels Hash of key-value pairs to be used
```
Comment (mwear) [inline sdk/lib/opentelemetry/sdk/resources/resource.rb]: I think we have been taking the easy way out by using `Hash<String, Object>`, but this is more accurate and an established convention according to the [yardoc site](https://yardoc.org/types#see_results). @fbogsany, should we go forward with this and change our other docs accordingly. Follow up question, if so, should we do that as part of this PR? Followup to the followup, eventually we'll be supporting array value attributes, which will level up the documentation complications, should we worry about this, or am I getting too far ahead.

Comment (fhwang) [reply sdk/lib/opentelemetry/sdk/resources/resource.rb]: If we want to update the other docs in the same way, I'm happy to do that next.

Comment (mwear) [reply sdk/lib/opentelemetry/sdk/resources/resource.rb]: That sounds good @fhwang. Do you want to do that as part of this PR or as a followup?

Also, taking a second look at this doc change, it does appear as though we have a lingering `>` and we should probably add spaces around the `=>`. Can we change this to `[Hash{String => String, Numeric, Boolean}]`?

Comment (fhwang) [reply sdk/lib/opentelemetry/sdk/resources/resource.rb]: I suppose I'll do that work as a followup PR, in the interest of keeping PRs small when possible.

I've made the requested changes.

**Pull request merged**